BepInEx.Core/Paths.cs

```diff
@@ -104,20 +104,31 @@ public static void SetExecutablePath(string executablePath,
 
         GameRootPath = PlatformHelper.Is(Platform.MacOS)
                            ? Utility.ParentDirectory(executablePath, 4)
-                           : Path.GetDirectoryName(executablePath);
+                           : Path.GetDirectoryName(executablePath) 
+                          ?? throw new DirectoryNotFoundException("Failed to extract GameRootPath from executablePath: " + executablePath);
```

Comment:
You are only applying this null check to `Path.GetDirectoryName` even though `Utility.ParentDirectory` will also return `null` in the same way. It will only return `null` for paths that don't have a parent like `/` or `C:/` so we are safe here though.
This should be tackled in a tree wide nullability PR not here.